feat(helm-v3): install custom plugin set

michelleN opened the pull request.

mattfarina commented on helm-v3/005-plugins.md:
```diff
@@ -63,8 +68,28 @@ used for executing Lua plugins mirrors the pattern for writing command-line
+## Installing Plugins
+A single plugin can be installed via `helm plugin install`.
+
+A set of plugins can also be defined in a yaml file and installed during the `helm init` process by using a flag called `--custom plugins <file.yaml>`
```
Shouldn't it be `--custom-plugins <file.yaml>` (notice the - between custom and plugins)?

michelleN replied:
You're right! Instead of fixing that though, I dropped the `custom` and named that flag `--plugins` instead. Let me know what you think.

mattfarina commented on helm-v3/005-plugins.md:
````diff
@@ -63,8 +68,28 @@ used for executing Lua plugins mirrors the pattern for writing command-line
+```
+plugins:
+- name: helm-template
+  version: 0.1.0
````
How will we resolve versions?

mattfarina replied:
@michelleN we discussed this in the developer call. Figuring out how to handle fetching the right version when the tip of master isn't the same as what was declared is difficult.

What about dropping the version info in the first go at this? This was suggested by @technosophos

technosophos commented:
I would suggest removing the version field for now, and I'd suggest this for the following reasons:

- There is no discovery protocol through which we could select which version to install because there is no plugin repository
- It means we have to write a version manager that can solve the question of "if the initial file says 2.0.0 and the _user_ wants to upgrade to 2.0.1, do we allow?"

If we end up having to solve this situation in future versions, we can do it then. But I think that at the outset it would just make the spec more difficult to implement than it needs to be.

That aside, I'm 👍 on this proposal.

michelleN commented:
@technosophos @mattfarina sure that makes sense to me. I dropped the `version` field in `file.yaml` and I also renamed the flag to `--plugins` instead of `--custom-plugins` for clarity and brevity. Let me know what ya'll think. Thanks for the input.
